soundcard/mediafoundation.py

```diff
@@ -758,7 +767,11 @@ def _record_chunk(self):
         self._idle_start_time = None
         data_ptr, nframes, flags = self._capture_buffer()
         if data_ptr != _ffi.NULL:
-            chunk = numpy.fromstring(_ffi.buffer(data_ptr, nframes*4*len(set(self.channelmap))), dtype='float32')
+            # Convert the raw CFFI buffer into a standard bytes object to ensure compatibility
+            # with modern NumPy versions (fromstring binary mode was removed). Using frombuffer
+            # on bytes plus .copy() guarantees a writable float32 array for downstream processing.
+            buf = bytes(_ffi.buffer(data_ptr, nframes * 4 * len(set(self.channelmap))))
+            chunk = numpy.frombuffer(buf, dtype=numpy.float32).copy()
```

Comment:
If I remember correctly, `frombuffer` is supposed to use buffer objects directly, so we can skip the `bytes` conversion (which also incurs another unnecessary copy).